hudi-spark-datasource/hudi-spark4-common/src/main/scala/org/apache/spark/sql/adapter/BaseSpark4Adapter.scala

```diff
@@ -186,4 +187,8 @@ abstract class BaseSpark4Adapter extends SparkAdapter with Logging {
       applyFiltersToPlan(logicalRelation, requiredSchema, resolvedSchema,
         relation.fileFormat.asInstanceOf[HoodieFormatTrait].getRequiredFilters))
   }
+
+  override def enableParquetFilterPushDownStringPredicate(conf: SQLConf): Boolean = {
+    conf.parquetFilterPushDownStringPredicate
```

Comment:
is the flag controled through Spark SQL and needs manual setup?

Reply:
> is the flag controled through Spark SQL and needs manual setup?

through Spark SQL